model/project_parser_db.go

```diff
@@ -114,17 +112,18 @@ func checkConfigNumberQuery(id string, configNum int) bson.M {
 	return q
 }
 
-// TryInsert suppresses the error of inserting if it's a duplicate key error and attempts to
-// upsert if config number matches
+// TryUpsert suppresses the error of inserting if it's a duplicate key error
+// and attempts to upsert if config number matches.
 func (pp *ParserProject) TryUpsert() error {
 	err := ParserProjectUpsertOne(checkConfigNumberQuery(pp.Id, pp.ConfigUpdateNumber), setAllFieldsUpdate(pp))
-	if isValidErr(err) {
+	if !db.IsDuplicateKey(err) {
```

Comment:
you should do errors.Cause()

Reply:
This is used all over the place without errors.Cause. I can move it into isDuplicateKey but I think it's messy to do it before.

Reply:
agreed, that it should move into isdupilicate key